wsmaster/che-core-sql-schema/src/main/resources/che-schema/7.0.0-beta3.0/1__add_devfile.sql

```diff
@@ -0,0 +1,260 @@
+    devfile_actions_id      BIGINT,
+
+    PRIMARY KEY (id)
+);
+ALTER TABLE devfile_action ADD CONSTRAINT fk_devfile_actions_id FOREIGN KEY (devfile_actions_id) REFERENCES devfile_command (id);
+CREATE INDEX index_action_command_id ON devfile_action (devfile_actions_id);
+
+
+-----------------------------------------------------------------------------------------------------------------------------------
+
+-- devfile component
+CREATE TABLE devfile_component (
+    id                     BIGINT       NOT NULL,
+    name                   VARCHAR(255) NOT NULL,
+    type                   VARCHAR(255) NOT NULL,
+    component_id           TEXT,
+    reference              TEXT,
+    reference_content      TEXT,
+    image                  TEXT,
+    memory_limit           VARCHAR(255),
+    mount_sources          BOOLEAN,
+    devfile_components_id  BIGINT,
+
+    PRIMARY KEY (id)
+);
+
+ALTER TABLE devfile_component ADD CONSTRAINT fk_devfile_component_id FOREIGN KEY (devfile_components_id) REFERENCES devfile (id);
+CREATE INDEX index_component_devfile_id ON devfile_component (devfile_components_id);
+
+---------
+
+CREATE TABLE component_command (
+    devfile_component_id    BIGINT,
+    commands                VARCHAR(255)
+);
+
+--constraints
+ALTER TABLE component_command ADD CONSTRAINT fk_component_command_component_id FOREIGN KEY (devfile_component_id) REFERENCES devfile_component (id);
+CREATE INDEX index_commands_component_id ON component_command (devfile_component_id);
+
+CREATE TABLE component_arg (
+    devfile_component_id    BIGINT,
+    args                    VARCHAR(255)
+);
+
+--constraints
+ALTER TABLE component_arg ADD CONSTRAINT fk_component_command_arg_id FOREIGN KEY (devfile_component_id) REFERENCES devfile_component (id);
+CREATE INDEX index_args_component_id ON component_arg (devfile_component_id);
+---------
+-- component selector
+CREATE TABLE component_selector (
+    component_id    BIGINT,
+    selector_key    VARCHAR(255),
+    selector        VARCHAR(255)
+);
+
+--constraints
+ALTER TABLE component_selector ADD CONSTRAINT fk_component_selector_id FOREIGN KEY (component_id) REFERENCES devfile_component (id);
+CREATE INDEX index_component_selector_entrypoint_id ON component_selector (component_id);
+
+---------
+
+-- devfile endpoint
+CREATE TABLE devfile_endpoint (
+    id                      BIGINT       NOT NULL,
+    name                    VARCHAR(255) NOT NULL,
+    port                    INTEGER      NOT NULL,
+    devfile_endpoints_id    BIGINT,
+
+    PRIMARY KEY (id)
+);
+ALTER TABLE devfile_endpoint ADD CONSTRAINT fk_devfile_endpoint_id FOREIGN KEY (devfile_endpoints_id) REFERENCES devfile_component (id);
+CREATE INDEX index_devfile_endpoint_id ON devfile_endpoint (devfile_endpoints_id);
+
+-- devfile endpoint attributes
+CREATE TABLE devfile_endpoint_attributes (
+    endpoint_id    BIGINT,
+    value          TEXT,
+    name           VARCHAR(255)
+);
+-- constraints
```

Comment:
Some `constraints` sections have such comment some not. Please do in the same way

Reply:
Also index is not a constraint, is it?

Reply:
aligned